pkg/resources/csi/vsphere/secret.go

```diff
@@ -50,7 +51,7 @@ func CloudConfigSecretNameReconciler(data *resources.TemplateData) reconciling.N
 				return nil, err
 			}
 
-			cloudConfig, err := CloudConfigCSIToString(vsphereCloudConfig)
+			cloudConfig, err := vsphere.CloudConfigToString(vsphereCloudConfig)
```

Comment:
Is this using the regular cloud-config? cloud-config for the CCM and the CSI is different, hence the `CloudConfigCSIToString` function. I don't think we should get rid of this function.